kv/union_store.go

```diff
@@ -39,6 +39,23 @@ type UnionStore interface {
 	GetMemBuffer() MemBuffer
 }
 
+// ContractType is the type of a contract
```

Comment:
I don't get what the contract means here.

Reply:
I think precondition is better.

Reply:
There is a subtle difference, precondition in https://github.com/pingcap/tidb/pull/9127 is something that could fail,
While the contract here is something that must hold.

I reuse the kvproto structure, but not yet reuse the code because they are different.
Should we use the same code?